### PR TITLE
Update Vim to 8.2.5172 to fix CVE-2022-2175, CVE-2022-2182

### PR DESCRIPTION
**Pull request by dmcilvaney**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 
Bump Vim minor version to 8.2.5172 to fix  CVE-2022-2175, CVE-2022-2182.

###### Change Log  

- Bump Vim minor version to 8.2.5172 to fix  CVE-2022-2175, CVE-2022-2182.

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2022-2175

###### Test Methodology

- Local build, tested in container

### CODE REVIEW

File: SPECS/vim/vim.signatures.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
  "Signatures": {
-  "vim-8.2.5154.tar.gz": "43bc45d305cf93fb4ba8f6619c7ac02209daee1f8f3ae071c2dbd635d9f587e3"
+  "vim-8.2.5154.tar.gz": "366a45168f59d7218ea22597432e64ac05c2626703b5d8fa70f37bfd674cd2c1"
```
Comment (oliviacrain) [inline SPECS/vim/vim.signatures.json:3]: Filename here is incorrect

File: cgmanifest.json [hunk 2]
```diff
@@ -26148,7 +26148,7 @@
         "other": {
           "name": "vim",
           "version": "8.2.5154",
```
Comment (oliviacrain) [inline cgmanifest.json:26150]: Version needs to be updated here

File: SPECS/vim/vim.signatures.json (hunk 1 above)
Comment (dmcilvaney) [reply SPECS/vim/vim.signatures.json]: Had my build environment fall over, forgot to redo this change. Thanks for catching this one.

File: cgmanifest.json (hunk 2 above)
Comment (dmcilvaney) [reply cgmanifest.json]: Have a commit comming to fix this

**Pull request merged**